### PR TITLE
docs: describe full bootstrapper flow

### PR DESCRIPTION
**Pull request by lucab**

### CODE REVIEW

**Comment by lucab** [comment]
@coresolve @colemickens if you have few spare minutes, I'd like to hear if this answers the high-level question of "how the bootstrapper works" or if you have other unanswered doubts after reading this.

File: Documentation/bootstrapper.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+```
+Template variables are replaced with node-specific values. A detached signature is provided at the same URL suffixed with a `.asc` extension.
+
+## Sources of information
+
+Bootstrapper tries to gather state from the cluster and from a [remote bucket][remote], in order to prepare an up-to-date Kubernetes node.
+
+This is the logic flow and the information sources it queries:
+ 1. Get cluster version from api-server `/version` endpoint
+ 1. Iff previous step permanently failed, use `/etc/kubernetes/installer/kubelet.env` to determine original cluster version (from installation time)
```
Comment (squeed) [inline Documentation/bootstrapper.md]: I would add some kind of explicit description that this is needed for bootstrap, since there will be no api server.

**Comment by squeed** [comment]
Would you like me to outline the version selection logic?

File: Documentation/bootstrapper.md (hunk 1 above)
Comment (lucab) [reply Documentation/bootstrapper.md]: Yep. I'll add few sentences to better describe node bootstrap in cluster bootstrap vs existing cluster scenario.

File: Documentation/bootstrapper.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ 1. Get runtime mappings from `tectonic-torcx-runtime-mappings` ConfigMap
+ 1. Iff previous step permanently failed, use `/etc/kubernetes/installer/runtime-mappings.yaml` to determine runtime mappings
```
Comment (colemickens) [inline Documentation/bootstrapper.md]: I thought it used the `/versions.yaml` file from this repo / the container image. Or is that changing right now, or was I just mistaken?

Comment (lucab) [reply Documentation/bootstrapper.md]: It does at the moment, yes. This was trying to describe the final approach, but I agree it is better to document the status quo and change it later.

File: Documentation/bootstrapper.md (hunk 1 above)
Comment (lucab) [reply Documentation/bootstrapper.md]: Added at the bottom of this numbered list.

File: Documentation/bootstrapper.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+The bootstrapper runs on every new Tectonic node, at most once, before the node can join the cluster.
+It must write the `/etc/kubernetes/kubelet.env` in order for the kubelet to start.
+It is deployed via Ignition as a systemd service called `k8s-node-bootstrap`.
+This service is installed and owned by tectonic-installer via the relevant [service unit][bootstrap-service].
+
+## Configuration knobs
+
+A few configuration options are available on the command-line. These are the most interesting:
+ * `--upgrade-os=<bool>`: whether to check for new OS updates. Defaults to `true`
```
Comment (squeed) [inline Documentation/bootstrapper.md]: "whether to check for and install OS updates"

**Pull request merged**